upcloud_api/server.py

```diff
@@ -505,28 +504,35 @@ def get_ip(self, access='public', addr_family=None, strict=None):
         if addr_family not in ['IPv4', 'IPv6', None]:
             raise Exception("`addr_family` must be 'IPv4', 'IPv6' or None")
 
-        if access not in ['private', 'public']:
-            raise Exception("`access` must be 'public' or 'private'")
+        if access not in ['private', 'public', 'utility']:
+            raise Exception("`access` must be 'public', 'utility' or 'private'")
 
-        if not hasattr(self, 'ip_addresses'):
-            self.populate()
+        if not hasattr(self, 'networking'):
+            raise Exception(
+                "`networking` attribute is missing, server details must be fetched first"
+            )
 
-        # server can have several public or private IPs
-        ip_addrs = [ip_addr for ip_addr in self.ip_addresses if ip_addr.access == access]
+        ip_addrs = []
+        for iface in self.networking['interfaces']['interface']:
+            ip_addresses = iface['ip_addresses']['ip_address']
+            if len(ip_addresses) == 0:
+                continue
 
-        # prefer addr_family (or IPv4 if none given)
-        preferred_family = addr_family if addr_family else 'IPv4'
```

Comment:
Is this a breaking change? As the get_ip() will behave differently now 🤔 At least should update the method documentation for "ip_addr_family" at the top to mention that None returns all :blush:

Reply:
We're still returning IPv6 address if addr_family is not defined. I guess we could make an extra check if we have a v4 address available in the possible IPs so interface ordering does not matter, I'll add that.

Reply:
Yeah was just thinking if someone has code like `ip_address = get_ip()[0]` and server has 1 ipv4 + 1 ipv6, previously they would get only ipv4 guaranteed, but now i guess they might get list of 2 with ipv6 in front 🤔

Reply:
Changed to prefer IPv4 when no IP family has been defined.